lib/authentication/service/auth_provider.dart

```diff
@@ -184,6 +148,38 @@ class AuthProvider with ChangeNotifier {
     return _firebaseUser.uid;
   }
 
+  Future<void> _emigrate(Map<String,dynamic> data) async{
```

Comment:
No one's emigrating here. Call the method "migrateToNewClassFormat" and the argument "userData". Add some Dartdoc to explain the old format as opposed to the new one.